plugins/inputs/solr/solr.go

```diff
@@ -0,0 +1,424 @@
+			"95th_pc_request_time":       metrics.Stats.Nine5thPcRequestTime,
+			"999th_pc_request_time":      metrics.Stats.Nine99thPcRequestTime,
+			"99th_pc_request_time":       metrics.Stats.Nine9thPcRequestTime,
+			"avg_requests_per_second":    metrics.Stats.AvgRequestsPerSecond,
+			"avg_time_per_request":       metrics.Stats.AvgTimePerRequest,
+			"errors":                     metrics.Stats.Errors,
+			"handler_start":              metrics.Stats.HandlerStart,
+			"median_request_time":        metrics.Stats.MedianRequestTime,
+			"requests":                   metrics.Stats.Requests,
+			"timeouts":                   metrics.Stats.Timeouts,
+			"total_time":                 metrics.Stats.TotalTime,
+		}
+		acc.AddFields(
+			fmt.Sprintf("solr_%s", strings.ToLower(category)),
+			coreFields,
+			map[string]string{
+				"core":    core,
+				"handler": name},
+			measurementTime,
+		)
+	}
+	return nil
+}
+
+func gatherUpdateHandlerMetrics(mbeansJSON json.RawMessage, core, category string, acc telegraf.Accumulator) error {
+	var coreMetrics map[string]UpdateHandler
+
+	measurementTime := time.Now()
+	if err := json.Unmarshal(mbeansJSON, &coreMetrics); err != nil {
+		return err
+	}
+	for name, metrics := range coreMetrics {
+		var autoCommitMaxTime int
+		if len(metrics.Stats.AutocommitMaxTime) > 2 {
+			autoCommitMaxTime, _ = strconv.Atoi(metrics.Stats.AutocommitMaxTime[:len(metrics.Stats.AutocommitMaxTime)-2])
+		}
+		coreFields := map[string]interface{}{
+			"class_name":                  metrics.Class,
+			"adds":                        metrics.Stats.Adds,
+			"autocommit_max_docs":         metrics.Stats.AutocommitMaxDocs,
+			"autocommit_max_time":         autoCommitMaxTime,
+			"autocommits":                 metrics.Stats.Autocommits,
+			"commits":                     metrics.Stats.Commits,
+			"cumulative_adds":             metrics.Stats.CumulativeAdds,
+			"cumulative_deletes_by_id":    metrics.Stats.CumulativeDeletesByID,
+			"cumulative_deletes_by_query": metrics.Stats.CumulativeDeletesByQuery,
+			"cumulative_errors":           metrics.Stats.CumulativeErrors,
+			"deletes_by_id":               metrics.Stats.DeletesByID,
+			"deletes_by_query":            metrics.Stats.DeletesByQuery,
+			"docs_pending":                metrics.Stats.DocsPending,
+			"errors":                      metrics.Stats.Errors,
+			"expunge_deletes":             metrics.Stats.ExpungeDeletes,
+			"optimizes":                   metrics.Stats.Optimizes,
+			"rollbacks":                   metrics.Stats.Rollbacks,
+			"soft_autocommits":            metrics.Stats.SoftAutocommits,
+		}
+		acc.AddFields(
+			fmt.Sprintf("solr_%s", strings.ToLower(category)),
+			coreFields,
+			map[string]string{
+				"core":    core,
+				"handler": name},
+			measurementTime,
+		)
+	}
+	return nil
+}
+
+func gatherCacheMetrics(mbeansJSON json.RawMessage, core, category string, acc telegraf.Accumulator) error {
+	var coreMetrics map[string]Cache
+
+	measurementTime := time.Now()
```

Comment:
All the metrics were retrieved at the exact same time. Instead of creating a separate timestamp for each measurement, I'd use the same timestamp for all. It'll make it a lot easier for users to correlate the metrics without having to deal with time windows/tolerances.